pkg/pagerduty/pagerduty.go

```diff
@@ -2,112 +2,154 @@ package pagerduty
-			} else if currentClusterID != prevClusterID {
-				return "", fmt.Errorf("not all alerts have the same cluster ID")
-			}
+			return nil, err
 		}
-		return prevClusterID, nil
+		alerts = append(alerts, formatAlert)
 	}
+	return alerts, nil
 }
 
-// GetClusterIDFromAlert extracts the cluster ID from a PagerDuty incident alert.
-// It expects the alert's body to have a Common Event Format.
-func (c *RealPagerDutyClient) GetClusterIDFromAlert(alert *pagerduty.IncidentAlert) (string, error) {
-	if alert == nil || alert.Body == nil {
-		return "", fmt.Errorf("given alert or it's body is empty")
-	}
+func (pd *PagerDuty) formatAlert(alert *pdApi.IncidentAlert) (formatAlert Alert, err error) {
+	formatAlert.IncidentID = alert.Incident.ID
 
-	cefDetails, ok := alert.Body["cef_details"].(map[string]interface{})
-	if !ok || cefDetails == nil {
-		return "", fmt.Errorf("missing or invalid Common Event Format Details of given alert")
-	}
+	formatAlert.Name = alert.Summary
+	formatAlert.Status = alert.Status
+	formatAlert.WebURL = alert.HTMLURL
 
-	detailsValue, ok := cefDetails["details"]
-	if !ok || detailsValue == nil {
-		return "", fmt.Errorf("missing or invalid 'details' field in Common Event Format Details")
-	}
+	// Check if the alert is of type 'Missing cluster'
+	isCHGM := alert.Body["details"].(map[string]interface{})["notes"]
 
-	details, ok := detailsValue.(map[string]interface{})
-	if !ok || details == nil {
-		return "", fmt.Errorf("'details' field is not a map[string]interface{} in Common Event Format Details")
+	if isCHGM != nil {
+		notes := strings.Split(fmt.Sprint(alert.Body["details"].(map[string]interface{})["notes"]), "\n")
+
+		formatAlert.ClusterID = strings.Replace(notes[0], "cluster_id: ", "", 1)
+		formatAlert.ClusterName = strings.Split(fmt.Sprint(alert.Body["details"].(map[string]interface{})["name"]), ".")[0]
+
+	} else {
+		formatAlert.ClusterID = fmt.Sprint(alert.Body["details"].(map[string]interface{})["cluster_id"])
+		formatAlert.ClusterName, err = pd.GetClusterName(alert.Service.ID)
+
+		// If the service mapped to the current incident is not available (404)
+		if err != nil {
+			formatAlert.ClusterName = "N/A"
+		}
 	}
 
-	clusterID, ok := details["cluster_id"].(string)
-	if !ok || clusterID == "" {
-		return "", fmt.Errorf("missing or invalid 'cluster_id' field in CEF details")
+	// If there's no cluster ID related to the given alert
+	if formatAlert.ClusterID == "" {
+		formatAlert.ClusterID = "N/A"
 	}
-	return clusterID, nil
+
+	return formatAlert, nil
 }
 
-// GetClusterID retrieves the cluster ID associated with the given incident ID.
-func (c *RealPagerDutyClient) GetClusterID(incidentID string) (string, error) {
-	incidentAlerts, err := c.PdClient.ListIncidentAlertsWithContext(context.TODO(), incidentID, pagerduty.ListIncidentAlertsOptions{})
+// GetClusterName interacts with the PD service endpoint and returns the cluster name string.
+func (pd *PagerDuty) GetClusterName(serviceID string) (string, error) {
```

Comment:
Why do we need the cluster name?

Reply:
I may need cluster name when automating the elevate reason via cluster login